cvefeed/nvd/match_cpe.go

```diff
@@ -0,0 +1,94 @@
+package nvd
+
+import (
+	"fmt"
+
+	"github.com/facebookincubator/nvdtools/cvefeed/nvd/schema"
+	"github.com/facebookincubator/nvdtools/wfn"
+)
+
+// cpeMatch is a wrapper around the actual NVDCVEFeedJSON10DefCPEMatch
+type cpeMatch struct {
+	*wfn.Attributes
+	versionEndExcluding   string
+	versionEndIncluding   string
+	versionStartExcluding string
+	versionStartIncluding string
+	hasVersionRanges      bool
+}
+
+// Matcher returns an object which knows how to match attributes
+func cpeMatcher(nvdMatch *schema.NVDCVEFeedJSON10DefCPEMatch) (wfn.Matcher, error) {
```

Comment:
@skogtwin please review the cpeMatcher and the Match function below more thoroughly, because that's the "meaty" part of the matching.